bftengine/tests/bcstatetransfer/source_selector_test.cpp

```diff
@@ -0,0 +1,241 @@
+// Concord
+//
+// Copyright (c) 2021 VMware, Inc. All Rights Reserved.
+//
+// This product is licensed to you under the Apache 2.0 license (the "License").
+// You may not use this product except in compliance with the Apache 2.0
+// License.
+//
+// This product may include a number of subcomponents with separate copyright
+// notices and license terms. Your use of these subcomponents is subject to the
+// terms and conditions of the subcomponent's license, as noted in the
+// LICENSE file.
+
+#include "gtest/gtest.h"
+
+#include "SourceSelector.hpp"
+#include "Logger.hpp"
+
+namespace {
+
+using bftEngine::bcst::impl::SourceSelector;
+using bftEngine::bcst::impl::NO_REPLICA;
+
+constexpr auto kRetransmissionTimeoutMs = 100;
+constexpr auto kReplicaReplacementTimeoutMs = 300;
+
+constexpr uint64_t kSampleCurrentTimeMs = 1000;
+constexpr auto kSmallDeltaTime = kSampleCurrentTimeMs + 1;
+constexpr auto kBigDeltaTime = 10 * kSampleCurrentTimeMs;
+constexpr uint16_t kSampleReplicaId = 1;
+
+const auto replicas = std::set<uint16_t>{1, 2, 3, 4};
```

Comment:
A thing that came to mind is that we use `replicas` as `other` replicas, meaning that for a cluster of 4 replicas, a more meaningful count of others is 3. Maybe we can change that set to 3 or 6 elements and call it `other_replicas`. What do you think?

Reply:
The set now includes only three replicas, considering that the cluster contains four replicas in total.